PhysicsTools/PatAlgos/python/producersHeavyIons/heavyIonJetSetup.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import division
+
+from RecoHI.HiJetAlgos.HiRecoPFJets_cff import akCs4PFJets
+from RecoHI.HiJetAlgos.HiGenJets_cff import ak5HiGenJets
+from RecoHI.HiJetAlgos.HiGenCleaner_cff import heavyIonCleanedGenJets
+from PhysicsTools.PatAlgos.producersHeavyIons.heavyIonJets_cff import *
+from PhysicsTools.PatAlgos.tools.helpers import *
+
+def setupHeavyIonJets(process, tag, radius, task):
+    addToProcessAndTask(
+        tag + 'Jets',
+        akCs4PFJets.clone(rParam = radius / 10),
+        process,
+        task)
+
+    genjetcollection = 'ak' + str(radius) + 'HiGenJets'
+
+    addToProcessAndTask(
+        genjetcollection,
+        ak5HiGenJets.clone(rParam = radius / 10),
+        process,
+        task)
+
+    addToProcessAndTask(
+        'ak' + str(radius) + 'HiCleanedGenJets',
+        heavyIonCleanedGenJets.clone(src = genjetcollection),
+        process,
+        task)
```

Comment:
Yes, akCs4PFJets are already in AOD.
In the case of the 2018 PbPb data, the algorithm has evolved quite a bit.  In our private repo, we updated the name to read akFlowPuCs4PFJets, in order to make the distinction.
As of #30057 , akFlowPuCs has become the default, and the original algo will no longer be used, so I figured it was best to keep the name compact.
In the frontier release the Cs jets would be duplicated in AOD and miniAOD.  However, once the calibrated reaction plane becomes available in the miniAOD #31129 , I would like for the jet clustering to use this more precise determination, instead of the current fitting strategy (IIRC this amounts to simply flipping a flag).  The idea would be that AOD is produced promptly, with Cs jets based on a rough estimate of the event plane (or even with the flow modulated turned off),  then the calibrations are derived, and the miniAOD is produced with a small delay, with the best quality UE subtraction.   This is our tentative strategy for upcoming runs (2022+) though.

So given this situation:
a) would you prefer to use a different name for the "flow-modulated" Cs jets to make a distinction to the old algo from 2018?
 b) would you still like the reclustering of the jets to be handled as a special case, even though our baseline strategy is to always recluster the jets?

Reply:
Regarding the gen stuff, you propose to simply move it to a separate python file?

Reply:
> Regarding the gen stuff, you propose to simply move it to a separate python file?

separate file or a separate function

Reply:
> Yes, akCs4PFJets are already in AOD.
> In the case of the 2018 PbPb data, the algorithm has evolved quite a bit. In our private repo, we updated the name to read akFlowPuCs4PFJets, in order to make the distinction.
> As of #30057 , akFlowPuCs has become the default, and the original algo will no longer be used, so I figured it was best to keep the name compact.
> In the frontier release the Cs jets would be duplicated in AOD and miniAOD. However, once the calibrated reaction plane becomes available in the miniAOD #31129 , I would like for the jet clustering to use this more precise determination, instead of the current fitting strategy (IIRC this amounts to simply flipping a flag). The idea would be that AOD is produced promptly, with Cs jets based on a rough estimate of the event plane (or even with the flow modulated turned off), then the calibrations are derived, and the miniAOD is produced with a small delay, with the best quality UE subtraction. This is our tentative strategy for upcoming runs (2022+) though.
> 
> So given this situation:
> a) would you prefer to use a different name for the "flow-modulated" Cs jets to make a distinction to the old algo from 2018?
> b) would you still like the reclustering of the jets to be handled as a special case, even though our baseline strategy is to always recluster the jets?

Sorry, I probably missed the point about the name of the collections. If the collection available in AOD needs to be remade, I guess it's practical to keep the same name and just remake it *when needed*.

the description sounds a bit similar to what's done in pp: 
both AOD and miniAOD are made in the first pass, and then some time later there is a re-miniAOD campaign. This in fact works out even for MC (while in your case, it's unclear if RP recalibration can be really unknown).

From your description it sounds like the intention is to not provide miniAOD initially, which I would expect to introduce an extra burden for early test and validation (especially analysis-like) to have to depend on AOD. It may be more productive to have a model similar to the pp setup: there is a default miniAOD which can be/is made together with RECO/AOD, and then there is a re-miniAOD step. 
`pp_on_AA_2018` is a re-mini, while `pp_on_PbPb_run3` looks more like the initial setup.

Reply:
Ok, we can revisit whether we should have the miniAOD produced promptly.  So would you propose that I add a flag corresponding to the option to redo the jet clustering, and enable it for the pp_on_AA_2018, while not doing so or pp_on_PbPb_run3?   We can always update the run 3 strategy later.

Reply:
is this part of the discussion really limited to just the RP recalibration or are there more modules to be added on top of the 2018 AOD?
perhaps you can list the modules, if it's more than just about reading AOD and then running `akCs4PFJets` on top of it

Reply:
I think this discussion is really limited to redoing the jet clustering.  The rest of miniAOD really consists of associated various objects via PAT, slimming, and adding formerly analysis-level stuff like centrality, reaction plane, photon isolation, etc.

Reply:
does the recalibration enter as a part of ES/calibration during running of `akCs4PFJets`?

To be practical, if this is the only thing planned to be (regularly) rerun for AOD inputs and if the cost is below 1% of AOD time, we could have a uniform setup.